File: backend/models/Assessment.js
import mongoose from 'mongoose';

const assessmentSchema = new mongoose.Schema({
  candidateId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
  title: {
    type: String,
    required: [true, 'Assessment title is required'],
  },
  type: {
    type: String,
    enum: ['coding', 'mcq'],
    default: 'coding',
  },
  score: {
    type: Number,
    required: true,
  },
  maxScore: {
    type: Number,
    default: 100,
  },
  code: {
    type: String,
    default: '',
  },
  language: {
    type: String,
    default: 'javascript',
  },
  submittedAt: {
    type: Date,
    default: Date.now,
  },
}, {
  timestamps: true,
});

assessmentSchema.index({ candidateId: 1, submittedAt: -1 });

const Assessment = mongoose.model('Assessment', assessmentSchema);
export default Assessment;

File: backend-spring/src/main/java/com/learn2hire/model/Assessment.java
package com.learn2hire.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.index.CompoundIndex;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "assessments")
@CompoundIndex(name = "candidateId_submittedAt", def = "{'candidateId': 1, 'submittedAt': -1}")
public class Assessment {

    @Id
    @JsonProperty("_id")
    private String id;

    private String candidateId; // User ID reference

    private String title;

    @Builder.Default
    private String type = "coding"; // coding, mcq

    private Integer score;

    @Builder.Default
    private Integer maxScore = 100;

    @Builder.Default
    private String code = "";

    @Builder.Default
    private String language = "javascript";

    @Builder.Default
    private Instant submittedAt = Instant.now();

    @CreatedDate
    private Instant createdAt;

    @LastModifiedDate
    private Instant updatedAt;
}
